Include/object.h

```diff
@@ -536,11 +536,20 @@ PyAPI_FUNC(int) PyObject_SetAttr(PyObject *, PyObject *, PyObject *);
 PyAPI_FUNC(int) PyObject_HasAttr(PyObject *, PyObject *);
 #ifndef Py_LIMITED_API
 PyAPI_FUNC(int) _PyObject_IsAbstract(PyObject *);
-/* Same as PyObject_GetAttr(), but don't raise AttributeError. */
-PyAPI_FUNC(PyObject *) _PyObject_GetAttrWithoutError(PyObject *, PyObject *);
 PyAPI_FUNC(PyObject *) _PyObject_GetAttrId(PyObject *, struct _Py_Identifier *);
 PyAPI_FUNC(int) _PyObject_SetAttrId(PyObject *, struct _Py_Identifier *, PyObject *);
 PyAPI_FUNC(int) _PyObject_HasAttrId(PyObject *, struct _Py_Identifier *);
+/* Replacements of PyObject_GetAttr() and _PyObject_GetAttrId() which
+   don't raise AttributeError.
+
+   Return 1 and set *result != NULL if an attribute is found.
+   Return 0 and set *result == NULL if an attribute is not found;
+   an AttributeError is silenced.
+   Return -1 and set *result == NULL if an error other than AttributeError
+   is raised.
+*/
+PyAPI_FUNC(int) _PyObject_LookupAttr(PyObject *, PyObject *, PyObject **);
```

Comment:
I would rename the method to `PyObject_LookupAttr` (no leading underscore).  I'd use it in my libraries and the leading underscore would make me think that I shouldn't.

I also like Inada-san's suggestion to simplify the return value even further:

* `-1`, if an *unrelated* error occurred during attribute lookup
* `0` if no errors have occurred; `*result` will be set to `NULL` if the attribute was not found, and to a non-`NULL` value if it was found.

Reply:
I think this API is experimental and should be kept private for some time. We are still discussing it, and after making some design decision we may change it in future.

Reply:
I agree with @serhiy-storchaka.
While I feel this API is nice, I'm not sure we should set
`*result=NULL` when returning -1 or not.

Reply:
Don't we care C API methods with a leading underscore the same as without?  I've never seen us breaking the `_` prefixed API to be honest. So leading underscore or not it's set in stone.

Reply:
@1st1 Actually speaking, I broke `_PyObject_GenericGetAttrWithDict()`.  I added one argument to suppress AttributeError.

Reply:
Fair enough. The PR is LGTM.